util_codes/View.py

```diff
@@ -0,0 +1,1128 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib
+matplotlib.rcParams["font.size"] = 13
+
+def phase(z):
+#     val = np.arctan2(z.real, z.imag)
+    val = np.rad2deg(np.unwrap(np.angle((z))))
+    return val
+
+class DataView(object):
+    """
+        Provides viewingtions for Data
+        This can be inherited by XXX
+    """
+    def __init__(self):
+        pass
+
+    def set_xyz(self, x, y, z, normal="Z"):
+        self.normal = normal
+        if normal =="X" or normal=="x":
+            self.x, self.y, self.z = x, y, z
+            self.ncx, self.ncy, self.ncz = 1, y.size, z.size
+            self.Y, self.Z = np.meshgrid(y, z)
+            self.xyz = np.c_[x*np.ones(self.ncy*self.ncz), self.Y.flatten(), self.Z.flatten()]
+
+        elif normal =="Y" or normal =="y":
+            self.x, self.y, self.z = x, y, z
+            self.ncx, self.ncy, self.ncz = x.size, 1, z.size
+            self.X, self.Z = np.meshgrid(x, z)
+            self.xyz = np.c_[self.X.flatten(), y*np.ones(self.ncx*self.ncz), self.Z.flatten()]
+
+        elif normal =="Z" or normal =="z":
+            self.x, self.y, self.z = x, y, z
+            self.ncx, self.ncy, self.ncz = x.size, y.size, 1
+            self.X, self.Y = np.meshgrid(x, y)
+            self.xyz = np.c_[self.X.flatten(), self.Y.flatten(), z*np.ones(self.ncx*self.ncy)]
+
+    def eval_loc(self, srcLoc,obsLoc, sigvec, fvec, orientation, func):
+        self.srcLoc=srcLoc
+        self.obsLoc = obsLoc
+        self.sigvec = sigvec
+        self.fvec = fvec
```

Comment:
it is a np.array of frequencies. Some of the 1D plots evaluates components of the E-field at a particular location and display theirs values for a range of frequencies. Same with sigvec: Range of conductivity.